### ISSUE TITLE
torch.nn.BatchNorm1d hits internal assert with complex32 input

### ISSUE THREAD

**Issue by TTZTTZ1** [pytorch/pytorch#193825]
### 🐛 Describe the bug

`torch.nn.BatchNorm1d` hits an internal assertion failure when given a `complex32` input tensor.

The issue occurs with an unsupported dtype/input combination and terminates with an internal PyTorch assert instead of raising a normal user-facing exception.

Minimal reproduction:

```python
import torch

torch.nn.BatchNorm1d(1)(
    torch.zeros(2, 0, dtype=torch.complex32)
)
```

Actual behavior:

PyTorch terminates with:

```
false INTERNAL ASSERT FAILED at "/pytorch/aten/src/ATen/AccumulateType.cpp":23, please report a bug to PyTorch.
Unrecognized ScalarType: ComplexHalf
```

Expected behavior:

PyTorch should either support this dtype/input combination correctly or reject it with a clear Python-level exception instead of triggering an internal assertion.

## Environment

PyTorch version: 2.11.0+cu130

Is debug build: False

CUDA used to build PyTorch: 13.0

ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.5 LTS (x86_64)

Python version: 3.10.20

CUDA runtime version: 13.0.88

GPU:
NVIDIA GeForce RTX 5090

NVIDIA driver version:
580.105.08

CPU:
Intel(R) Xeon(R) Platinum 8470Q

Relevant libraries:

torch==2.11.0+cu130

triton==3.6.0

numpy==2.2.6

cc @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki @malfet @ezyang @anjali411 @dylanbespalko @nikitaved @amjames
